src/mlpack/methods/ann/layer/layer_types.hpp

```diff
@@ -92,6 +109,7 @@ using LayerTypes = boost::variant<
     DropConnect<arma::mat, arma::mat>*,
     Dropout<arma::mat, arma::mat>*,
     ELU<arma::mat, arma::mat>*,
+    FFN<NegativeLogLikelihood<>, RandomInitialization>*,
```

Comment:
Ohh thats not the case. So the template parameters passed on to the FFN are irrelevant. They won;t be used at all. The initialization is done by the base class, which in this case is the RNN in which NTM layer will be added. There the user can specify the appropriate initialization and error function. Both those template arguments, for this use of FFN as a controller are dummies.